lib/specValidator.js

```diff
@@ -46,17 +57,37 @@ class SpecValidator {
     this.specResolver = null;
     this.specValidationResult = { validityStatus: true, operations: {} };
     this.swaggerApi = null;
+    if (options) {
+      if (options.shouldResolveRelativePaths === null || options.shouldResolveRelativePaths === undefined) {
+        options.shouldResolveRelativePaths = true;
+      }
+      if (options.shouldResolveXmsExamples === null || options.shouldResolveXmsExamples === undefined) {
+        options.shouldResolveXmsExamples = true;
+      }
+      if (options.shouldResolveAllOf === null || options.shouldResolveAllOf === undefined) {
+        options.shouldResolveAllOf = true;
+      }
+      if (options.shouldSetAdditionalPropertiesFalse === null || options.shouldSetAdditionalPropertiesFalse === undefined) {
+        options.shouldSetAdditionalPropertiesFalse = true;
+      }
+      if (options.shouldResolvePureObjects === null || options.shouldResolvePureObjects === undefined) {
+        options.shouldResolvePureObjects = true;
+      }
+    }
+    this.options = options;
   }
 
   /*
    * Initializes the spec validator. Resolves the spec on different counts using the SpecResolver and initializes the internal api validator.
    */
   initialize() {
     let self = this;
-    utils.clearCache();
+    if (self.options.shouldResolveRelativePaths) {
+      utils.clearCache();
```

Comment:
We don't need to clear the cache if relative paths are already resolved. This should help @olydis.